### PR TITLE
[SYCL] Set indirect access kernel property only if needed

### PR DESCRIPTION
**Pull request by romanovvlad**
Need to set indirect access kernel property only if a kernel captures
pointers.

### CODE REVIEW

File: sycl/source/detail/scheduler/commands.cpp [hunk 1]
```diff
@@ -1683,6 +1684,7 @@ pi_result ExecCGCommand::SetKernelParamsAndLaunch(
     case kernel_param_kind_t::kind_pointer: {
       Plugin.call<PiApiKind::piextKernelSetArgPointer>(Kernel, Arg.MIndex,
                                                        Arg.MSize, Arg.MPtr);
+      IndirectAccessesNeeded = true;
```
Comment (smaslov-intel) [inline sycl/source/detail/scheduler/commands.cpp:1687]: Can't we do even better than this? Can't we track in the kernel somehow that some other memory is accessed through this pointer indirectly? E.g. if it is a pointer to an integer, then we don't need to set this attribute, but only if it is a pointer to memory where another pointer is contained *and it is actually used*.

**Comment by jbrodman** [comment]
This is not a safe change.

We quickly discovered that users tend to hide pointers inside structs.  Doing so would never enable indirect access with this strategy and programs now don't work.

File: sycl/source/detail/scheduler/commands.cpp (hunk 1 above)
Comment (jbrodman) [reply sycl/source/detail/scheduler/commands.cpp:1687]: @smaslov-intel - It's not about indirect loads through pointers.

"Indirect here" is more descended from OpenCL SVM.  OpenCL SVM required the kernel to state *every* pointer it *might* touch inside a kernel.  "Indirect" here says "you can just use them without telling me everything you might use"

**Comment by romanovvlad** [comment]
> 
> 
> This is not a safe change.
> 
> We quickly discovered that users tend to hide pointers inside structs. Doing so would never enable indirect access with this strategy and programs now don't work.

It seems hiding pointers inside structs is not a problem anymore, but cases when user "temporary" convert a pointer a an integer, pass to a kernel, an convert back is a problem.

**Pull request closed without merging**